packages/react-app/components/TokenContract.tsx

```diff
@@ -0,0 +1,489 @@
+  const [newAddress, setNewAddress] = useState("");
+  const [tokenTransferValue, setTokenTransferValue] = useState("");
+
+  const contract = contractData
+    ? new kit.web3.eth.Contract(contractData.abi, contractData.address)
+    : null;
+
+  useEffect(() => {
+    if (contractData) {
+      setContractLink(`${network.explorer}/address/${contractData.address}`);
+    }
+  }, [network, contractData]);
+
+  const setToken = async () => {
+    try {
+      await performActions(async (kit) => {
+        const gasLimit = await contract.methods.store(tokenInput).estimateGas();
+
+        const result = await contract.methods
+          .store(tokenInput)
+          .send({ from: address, gasLimit });
+
+        console.log(result);
+
+        const variant = result.status == true ? "success" : "error";
+        const url = `${network.explorer}/tx/${result.transactionHash}`;
+        const action = (key) => (
+          <>
+            <Link href={url} target="_blank">
+              View in Explorer
+            </Link>
+            <Button
+              onClick={() => {
+                closeSnackbar(key);
+              }}
+            >
+              X
+            </Button>
+          </>
+        );
+        enqueueSnackbar("Transaction sent", {
+          variant,
+          action,
+        });
+      });
+    } catch (e) {
+      console.log(e);
+    }
+  };
+
+  const getTokenName = async () => {
+    const tokenName = await contract.methods.name().call();
+    setTokenName(tokenName);
+  };
+
+  const getTokenSymbol = async () => {
+    const tokenSymbol = await contract.methods.symbol().call();
+    setTokenSymbol(tokenSymbol);
+  };
+
+  const getTokenInitialSupply = async () => {
+    const tokenInitialSupply = await contract.methods
+      .initialSupply()
+      .call()
+      .then((result) => {
+        return ethers.utils.formatUnits(result, 18);
+      });
+    setTokenInitialSupply(tokenInitialSupply);
+  };
+
+  const getTokenDecimals = async () => {
+    const tokenDecimals = await contract.methods.decimals().call();
+    setTokenDecimals(tokenDecimals.toString());
+  };
+
+  const getTokenTotal = async () => {
+    const tokenTotal = await contract.methods.totalSupply().call();
+    const convertedTotal = ethers.utils.formatUnits(tokenTotal.toString(), 18);
+    setTokenTotal(convertedTotal);
+  };
+
+  const getBalance = async () => {
+    const tokenBalance = await contract.methods
+      .balance()
+      .call()
+      .then((result) => {
+        return ethers.utils.formatUnits(result, 18);
+      });
+    setTokenBalance(tokenBalance);
+  };
+
+  const getTokenBalanceOf = async (e) => {
+    e.preventDefault();
+    console.log(address);
+    const tokenBalanceOf = await contract.methods
+      .balanceOf(address)
+      .call()
+      .then((result) => {
+        return ethers.utils.formatUnits(result, 18);
+      });
+    setTokenBalanceOf(tokenBalanceOf);
+  };
+
+  const transferTokens = async (e) => {
```

Comment:
The syntax in this function is incorrect. Check how a transaction is sent in the StorageContract interface again and try to mimic what is done here 

https://github.com/celo-org/celo-progressive-dapp-starter/blob/64a59df17b13a95208950524865ae0f5740c651c/packages/react-app/components/StorageContract.tsx#L53